lib/sendserver.c

```diff
@@ -121,6 +187,32 @@ static int rc_pack_list(VALUE_PAIR * vp, char *secret, AUTH_HDR * auth)
 			total_length += padded_length + 2;
 
 			break;
+		case PW_CHAP_PASSWORD:
+                        /* Encrypt the password */
+
+                        /* Record the attribute length 
+                         * Type(1) + Length(1) + Chap-ident(1) + Chap-string(16) 
+                         * */
+                        *buf++ = CHAP_VALUE_LENGTH + 3;
+
+                        /* Get random chap identifier */
+                        chap_id = random() & 0xff;
```

Comment:
Do we really need a random value here? If not shouldn't we set zero or one there to simplify the logic?

If we need a random value we should re-use `rc_get_id()` so that we do not use `random()` uncontrolled all over the place (it's a pretty ugly call which doesn't really provide unpredictable output).